### PR TITLE
[WIP] Add names next to pointers

### PR DESCRIPTION
**Pull request by vjeux**
This implements the rendering and messaging across. Still need to do the UI to set the name.

Also, not really sure what's the best place to send the name and store it.

ETHDHs1UUAE49Pl

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -411,6 +413,7 @@ export class App extends React.Component<any, AppState> {
         payload: {
           socketID: this.socket.id,
           pointerCoords: payload.pointerCoords,
+          username: "vjeux",
```
Comment (petehunt) [inline src/components/App.tsx]: i think this is a fine way to share the username

File: src/renderer/renderScene.ts [hunk 2]
```diff
@@ -163,6 +163,7 @@ export function renderScene(
   // Paint remote pointers
   for (const clientId in sceneState.remotePointerViewportCoords) {
     let { x, y } = sceneState.remotePointerViewportCoords[clientId];
+    const username = sceneState.remotePointerUsernames[clientId];
```
Comment (petehunt) [inline src/renderer/renderScene.ts]: do you think it'd be better to store the username along with the x and y?

File: src/renderer/renderScene.ts [hunk 3]
```diff
@@ -196,6 +197,30 @@ export function renderScene(
...
     context.stroke();
+
+    if (!isOutOfBounds && username) {
+      const offsetX = x + width;
+      const offsetY = y + height;
+      const paddingHorizontal = 4;
+      const paddingVertical = 4;
+      const measure = context.measureText(username);
+      const measureHeight =
+        measure.actualBoundingBoxDescent + measure.actualBoundingBoxAscent;
```
Comment (petehunt) [inline src/renderer/renderScene.ts]: TIL

Comment (vjeux) [reply src/renderer/renderScene.ts]: Me too!

File: src/renderer/renderScene.ts (hunk 2 above)
Comment (vjeux) [reply src/renderer/renderScene.ts]: I honestly don't know...

**Comment by petehunt** [comment]
if you want to land it quickly, maybe autogenerate a name based on the socket id from a dictionary of adjectives and animals like gdocs does ("speedy ibex", "nerdy moose" etc)

File: src/renderer/renderScene.ts (hunk 2 above)
Comment (vjeux) [reply src/renderer/renderScene.ts]: I was thinking about what else we could store here, some ideas:
- mouse up/down status
- current cursor

**Comment by vjeux** [comment]
I'd love for someone (you?) to finish and ship it :p Using a dictionary seems fine.

**Comment by dwelle** [comment]
Btw, I made the `collaborators` a map of objects for this exact reason so we can store additional data into it. :+1:

**Comment by j-f1** [comment]
I made [a thing for combining a random adjective and noun](https://observablehq.com/@j-f1/random-phrase-generator) and my suggestion is to definitely go with a restricted word list as there are a _ton_ of words nobody ever uses.

**Comment by petehunt** [comment]
#971

**Pull request closed without merging**